sympy/tensor/array/array_comprehension.py

```diff
@@ -38,6 +37,10 @@ def __new__(cls, function, *symbols, **assumptions):
                               ' for the expression')
         arglist = [sympify(function)]
         arglist.extend(cls._check_limits_validity(function, symbols))
+        return cls._new(arglist, **assumptions)
+
+    @classmethod
+    def _new(cls, arglist, **assumptions):
```

Comment:
why has `ArrayComprehension` been modified?

Reply:
By doing this, `_new` can be reused in the `ArrayComprehensionMap`. So we can avoid repeting codes.